PowerShellGet/PSModule.psm1

```diff
@@ -8584,22 +8596,53 @@ function ValidateAndAdd-PSScriptInfoEntry
                             $KeyName = $script:Version
 
                             [Version]$Version = $null
-
-                            if([System.Version]::TryParse($Value, ([ref]$Version)))
+                            
+                            # Special case for prerelease string in version field, must pull out before parsing version
+                            if ($Value -match '-') 
                             {
-                                $Value = $Version                            
+                                [String]$PreRelease = $null
+
+                                $indexOfFirstHyphen = $Value.IndexOf('-')
+                                $versionSubstring = $Value.Substring(0, $indexOfFirstHyphen)
+                                $preReleaseSubstring = $Value.Substring($indexOfFirstHyphen+1)
+
+                                if( [System.Version]::TryParse($versionSubstring, ([ref]$Version)) -and 
+                                    (Validate-PreReleaseString -Version $versionSubstring -PreRelease $preReleaseSubstring -InfoObject $Value) )
+                                {
+                                    $Value = [String]$Version + '-' + $preReleaseSubstring
+                                }
+                                else
+                                {
+                                    $message = $LocalizedData.InvalidVersion -f ($Value)
+                                    ThrowError -ExceptionName "System.ArgumentException" `
+                                                -ExceptionMessage $message `
+                                                -ErrorId "InvalidVersion" `
+                                                -CallerPSCmdlet $CallerPSCmdlet `
+                                                -ErrorCategory InvalidArgument `
+                                                -ExceptionObject $Value
+                                    return
+                                }

```

Comment:
This else block is throwing the same error as in below else/else block. Consider changing the logic to have a single block for throwing the InvalidVersion exception.